### PR TITLE
Define models and relationships

### PR DESCRIPTION
**Pull request by ppmadalin**

### CODE REVIEW

**Comment by ppmadalin** [comment]
I am not sure if the UserAnswer needs a many-to-many relationship. Need your feedback on this.

File: app/models.py [hunk 1]
```diff
@@ -1,29 +1,87 @@
...
-    date_created = db.Column(db.DateTime, default=db.func.current_timestamp())
-    date_modified = db.Column(
-        db.DateTime,
-        default=db.func.current_timestamp(),
-        onupdate=db.func.current_timestamp(),
+    answer = db.Column(db.String(80), nullable=False)
+    questionid = db.Column(db.Integer, db.ForeignKey("question.id"), nullable=False)
+    correct_answer = db.Column(db.Boolean, nullable=False)
+
+    # Relationship one-to-many
```
[the remark this answers is not in the document]
Comment (ppmadalin) [reply app/models.py]: Yep. You are right. Now I figure out. 

- One answer can have only one question
- One question can have more then one answer

**Comment by code-monk08** [comment]
> I am not sure if the UserAnswer needs a many-to-many relationship. Need your feedback on this.

as far as I can see, for one **user answer** there can be many questions and for one question there can be many **user answer** as we are also considering questions with multiple/more than 1 correct answers in our db.

and hence many to many.

File: app/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (code-monk08) [reply app/models.py]: @ppmadalin for example 'car' is an answer for some certain question, it's not necessary that 'car' cannot be an answer for other questions too.. so don't you think that

one answer can have many questions.

**Comment by code-monk08** [comment]
> I am not sure if the UserAnswer needs a many-to-many relationship. Need your feedback on this.

as far as I can see, for one user answer there can be many questions inside db and for one question there can be many user answer as we are also considering questions with multiple/more than 1 correct answers and hence not many to many.

File: app/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bbelderbos) [reply app/models.py]: `questionid` in Answer implies one-to-many of question->answers which I think is best/easiest, no?

**Comment by code-monk08** [comment]
we need some more work on this right @ppmadalin ?

**Pull request merged**